test/e2e_node/cpu_manager_test.go

```diff
@@ -557,6 +559,92 @@ func runMultipleCPUContainersGuPod(ctx context.Context, f *framework.Framework)
 	waitForContainerRemoval(ctx, pod.Spec.Containers[1].Name, pod.Name, pod.Namespace)
 }
 
+func runCfsQuotaGuPods(ctx context.Context, f *framework.Framework, disabledCPUQuotaWithExclusiveCPUs bool) {
```

Comment:
Can you add a check on the pod level cgroup here as well to be fully sure that behavior also works as expected on that level as well?

Reply:
hmm from within a running container you'd need to do some weird workarounds to allow access to that cgroup and fetch the info. off top of my head to get the pod cgroup you'd need to: 

* hostpath mount `/sys/fs/cgrouop/kubepods.slice` to somewhere like `/kubepods-slice` going to have do some some unusual

to then determine the proper cgroup you'd need to know the container id and recurse up a directory. something like 
* add POD_NAME and POD_NAMESPACE as env var via downward api
* curl apiserver with SA token as cert, parse to find container id 
* `find` directory with corresponding id within `/kubepods-slice`
* requires corresponding role+rolebinding+sa associated with the given pod to allow querying pod info to obtain container id

---
1) is there any other way that we can think of that'd be easier?
2) are e2e tests allowed to do any of this? I recall reading somewhere that PSS are enforced which would restrict hostpath mounting entirely as an option?

Reply:
I think you can start by looking at other test, like eg. this one;
https://github.com/kubernetes/kubernetes/blob/02fd991c5b3d1b705658633a93f8d8aae2c694bb/test/e2e_node/hugepages_test.go#L70

Its been a long time since I worked closely with the e2e tests, so things might have changes in the past year.


But overall, given the importance of this code, I very much would prefer if we tested this properly to avoid regressions or unintended behavior.

Reply:
thanks! got it worked out - did hostpath mount of `/sys/fs/cgroup` plus downward api to surface pod uid makes it easy to find the pod's cpu.max 👍